models/task/stats.go

```diff
@@ -1,17 +1,50 @@
 package task
 
 import (
+	"time"
+
 	"github.com/juju/errors"
 	"github.com/loopfz/gadgeto/zesty"
 	"github.com/ovh/utask/db/pgjuju"
 	"github.com/ovh/utask/db/sqlgenerator"
+	"github.com/ovh/utask/pkg/now"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var (
+	validationTimes = promauto.NewSummaryVec(prometheus.SummaryOpts{Name: "utask_valid_times"}, []string{"template"})
+	completionTimes = promauto.NewSummaryVec(prometheus.SummaryOpts{Name: "utask_complete_times"}, []string{"template"})
+	executionTimes  = promauto.NewSummaryVec(prometheus.SummaryOpts{Name: "utask_exec_times"}, []string{"template"})
 )
 
 type stateCount struct {
 	State string  `db:"state"`
 	Count float64 `db:"state_count"`
 }
 
+// RegisterValidationTime computes the duration between the task creation and
+// the associated resolution's creation. This metric is then pushed to Prometheus.
+func RegisterValidationTime(templateName string, taskCreation time.Time) {
+	duration := now.Get().Sub(taskCreation).Seconds()
+	validationTimes.WithLabelValues(templateName).Observe(duration)
+}
+
+// RegisterTaskTime computes the execution duration and the complete duration
+// (from creation to completion) of a task. These metrics are then pushed to Prometheus.
+func RegisterTaskTime(templateName string, tackCreation, resCreation time.Time) {
+	currentTime := now.Get()
+
+	// Time taken since task creation
+	completeTime := currentTime.Sub(tackCreation).Seconds()
```

Comment:
```suggestion
func RegisterTaskTime(templateName string, taskCreation, resCreation time.Time) {
	currentTime := now.Get()

	// Time taken since task creation
	completeTime := currentTime.Sub(taskCreation).Seconds()
```